examples/5_secret_storage/secret_storage.go

```diff
@@ -0,0 +1,535 @@
+}
+
+func (sc *SecretKeyKVStoreAPI) PutEntryWithRevision(teamName string, namespace string, entryKey string, entryValue string, revision int) (result keybase1.KVPutResult, err error) {
+	var keyedValue map[string]string
+	if err = json.Unmarshal([]byte(entryValue), &keyedValue); err != nil {
+		return
+	}
+
+	keyedValue[KeyKey] = entryKey
+	bytes, err := json.Marshal(keyedValue)
+	if err != nil {
+		return
+	}
+
+	hmacEntryKey, err := sc.hmacKey(teamName, namespace, entryKey)
+	if err != nil {
+		return
+	}
+
+	result, err = sc.api.PutEntryWithRevision(teamName, namespace, hmacEntryKey, string(bytes), revision)
+	if err != nil {
+		return
+	}
+	result.EntryKey = entryKey
+	return
+}
+
+func (sc *SecretKeyKVStoreAPI) DeleteEntry(teamName string, namespace string, entryKey string) (result keybase1.KVDeleteEntryResult, err error) {
+	return sc.DeleteEntryWithRevision(teamName, namespace, entryKey, 0)
+}
+
+func (sc *SecretKeyKVStoreAPI) DeleteEntryWithRevision(teamName string, namespace string, entryKey string, revision int) (result keybase1.KVDeleteEntryResult, err error) {
+	hmacEntryKey, err := sc.hmacKey(teamName, namespace, entryKey)
+	if err != nil {
+		return
+	}
+	result, err = sc.api.DeleteEntryWithRevision(teamName, namespace, hmacEntryKey, revision)
+	if err != nil {
+		return
+	}
+	result.EntryKey = entryKey
+	return
+}
+
+func (sc *SecretKeyKVStoreAPI) GetEntry(teamName string, namespace string, entryKey string) (result keybase1.KVGetResult, err error) {
+	hmacEntryKey, err := sc.hmacKey(teamName, namespace, entryKey)
+	if err != nil {
+		return
+	}
+	result, err = sc.api.GetEntry(teamName, namespace, hmacEntryKey)
+	if err != nil {
+		return
+	}
+	result.EntryKey = entryKey
+	return
+}
+
+func (sc *SecretKeyKVStoreAPI) ListNamespaces(teamName string) (keybase1.KVListNamespaceResult, error) {
+	return sc.ListNamespaces(teamName)
+}
+
+func (sc *SecretKeyKVStoreAPI) ListEntryKeys(teamName string, namespace string) (result keybase1.KVListEntryResult, err error) {
+	keys, err := sc.api.ListEntryKeys(teamName, namespace)
+	if err != nil {
+		return
+	}
+	tmp := keys.EntryKeys[:0]
+	for _, e := range keys.EntryKeys {
+		if strings.HasPrefix(e.EntryKey, "_") {
+			continue
+		}
+
+		get, err := sc.api.GetEntry(teamName, namespace, e.EntryKey)
+		if err != nil {
+			return result, err
+		}
+
+		var keyedValue map[string]string
+		if err := json.Unmarshal([]byte(get.EntryValue), &keyedValue); err != nil {
+			return result, err
+		}
+		e.EntryKey = keyedValue[KeyKey]
+		tmp = append(tmp, e)
+	}
+	keys.EntryKeys = tmp
+	return keys, nil
+}
+
+// Wraps a KVStoreClient to expose methods to handle tool rentals.
+// Tries kvstore write actions with explicit revision numbers.
+// If it fails to write, it does a "get" and returns the get result.
+type RentalBotClient struct {
+	api kbchat.KVStoreAPI
+}
+
+func (r *RentalBotClient) Lookup(teamName string, tool string) (keybase1.KVGetResult, error) {
+	return r.api.GetEntry(teamName, Namespace, tool)
```

Comment:
Can you add a `NewRentalBotClient(namespace string) *RentalBotClient` function so `Namespace` can be stored on `RentalBotClient` instead of being hardcoded?